### PR TITLE
Throw 404 http exceptions on missing pages

### LINKED ISSUE #75: Throw a proper 404 exception when loading a non-existent page
Currently the main controller just displays a page with a message that the /page/whatever does not exist. We should actually throw an http_exception in that case, with a 404 error code.

### PR DESCRIPTION
**Pull request by VSEphpbb**
Fixes #75

### CODE REVIEW

File: controller/main_controller.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 namespace phpbb\pages\controller;
 
 use Symfony\Component\DependencyInjection\ContainerInterface;
+use \phpbb\exception\http_exception;
```
Comment (michaelcullum) [inline controller/main_controller.php]: `\phpbb\` ?

Comment (VSEphpbb) [inline controller/main_controller.php]: That's the full namespace

Comment (nickvergessen) [inline controller/main_controller.php]: @VSEphpbb remove the leading slash ;)

**Comment by VSEphpbb** [comment]
Updated

**Comment by michaelcullum** [comment]
Please fix your commit message to `Change fully qualified namespace to qualified namespace in use statement`

File: controller/main_controller.php [hunk 2]
```diff
@@ -64,38 +66,30 @@ public function display($route)
...
-
-		// Set the page title to use
-		$page_title = ($display) ? $display->get_title() : $this->user->lang('INFORMATION');
+		$page = $this->load_page_data($route);
 
 		// Assign the page data to template variables
 		$this->template->assign_vars(array(
-			'PAGE_TITLE'	=> $page_title,
-			'PAGE_CONTENT'	=> ($display) ? $display->get_content_for_display() : $this->user->lang('PAGE_NOT_AVAILABLE', $route),
+			'PAGE_TITLE'	=> $page->get_title(),
```
Comment (michaelcullum) [inline controller/main_controller.php]: Lets not use this twice, let's store this to a property (or variable with local scope) when we load page data then access that property/var?

**Comment by VSEphpbb** [comment]
Updated per @michaelcullum

**Pull request merged**